### PR TITLE
TST: catch exception for div/truediv operations between Timedelta and pd.NA

### LINKED ISSUE #54315: TST: div/truediv with PyArrow duration types
Looks like they implemented ``__truediv__`` for more types.

https://github.com/pandas-dev/pandas/actions/runs/5707046701/job/15463354529

### PR DESCRIPTION
**Pull request by droussea2001**
- [x] closes #54315
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [Not applicable] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [Not applicable] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

This PR add a test to catch ``TypeError`` raised by div and true div operations between ``Timedelta`` and ``pd.NA``.

### CODE REVIEW

File: doc/source/whatsnew/v3.0.0.rst [hunk 1]
```diff
@@ -362,6 +362,7 @@ Datetimelike
 Timedelta
 ^^^^^^^^^
 - Accuracy improvement in :meth:`Timedelta.to_pytimedelta` to round microseconds consistently for large nanosecond based Timedelta (:issue:`57841`)
+- Add a test for div and true div between ``Timedelta`` and ``pd.NA`` raising a TypeError (:issue:`54315`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:365]: Whatsnew notes are not needed for just unit test additions

Comment (droussea2001) [reply doc/source/whatsnew/v3.0.0.rst]: Thanks, I just removed it

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @droussea2001

**Comment by droussea2001** [comment]
You're welcome !

**Pull request merged**